### ISSUE TITLE
Test: file icon support per file type

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#59246]
Ref #4845

Complexity: 2

- [ ] Mac @kieferrm 
- [x] Windows @mjbvz

We now provide file icons per file type for the top 20 extensions. Verify you see these file icons when associating VSCode. On Windows, this requires a installation where the checkbox is checked to associate VSCode for file types. The icon will only show up when the OS explorer is configured to show medium-large icons. For small icons we decided to keep the VSCode logo.

**Note:** I noticed the OS acting up badly when a previous file association changes to a new one. I often felt like I needed to completely uninstall and reinstall or [even clear some caches](https://sixcolors.com/post/2015/10/clean-out-a-messy-open-with-menu-in-the-finder/). If you found a trick to make this work, let me know.

**Comment by Astrantia** [comment]
@bpasero With the latest insiders all icons are gone. Was the commit reverted?

**Comment by bpasero** [comment]
@Astrantia no, not that I am aware of, @misolori could maybe comment

**Comment by misolori** [comment]
@Astrantia the icons should show the vs code icon at small sizes and larger ones show the icon type with the vs code logo, like so:

<img width="50%" src="https://user-images.githubusercontent.com/35271042/46165478-91d53080-c245-11e8-96c5-943a856d7103.png">
<img width="50%" src="https://user-images.githubusercontent.com/35271042/46165479-939ef400-c245-11e8-8fb2-23e3b9dbcfad.png">

**Comment by kieferrm** [comment]
Not quite sure how to continue testing this. For some file types such as `.ts` I see the new icons for others I don't despite having changed the default app, cleared cached icons, completely uninstalled vscode, rebooted, ... Still seeing the Xcode icons for most file types.

**Comment by misolori** [comment]
I'm also seeing some wierdness on the mac where some icons only show up in small views (notice my Markdown and JS files):

![gif](https://user-images.githubusercontent.com/35271042/46167102-c3e89180-c249-11e8-8fba-732410fc43a7.gif)

@kieferrm is this what you're also seeing?

**Comment by kieferrm** [comment]
This is what I see, no matter the icon size:

![image](https://user-images.githubusercontent.com/4674940/46167545-e6c77580-c24a-11e8-94da-d086fb85005e.png)

**Comment by Astrantia** [comment]
@misolori @kieferrm I can confirm I see correct icons when resized **on Windows**.

**Comment by kieferrm** [comment]
I made a clean macOS install in parallels. I see the same behavior that @misolori describes. I.e. default icons size is the non-vscode icon. When I scale down to 28x28 the vs code icons start to show up.

**Comment by musm** [comment]
Thanks for adding this, but why can't we have custom file icons for all the sizes? Right
now it's pretty  useless since the file icon size has to be *massive* (at a size I never use) to be visible.now it's pretty useless since the file icon size has to be *massive* (at a size I never use) to be visible.

![f](https://user-images.githubusercontent.com/4319522/46627440-7f41dd80-cb08-11e8-8976-6c9302acd5a5.gif)

**Comment by misolori** [comment]
@musm the icons are harder to see in the small size, which is why we decided to go with the logo in the small view. Please feel free to open a new issue to track that request. Thanks!

**Comment by brezanac** [comment]
Windows 10 build 1809, VSCode 1.28.0 (user setup)

The file icons refused to display properly even though Visual Studio Code was set as the default application for all of them.

What helped in my case is going to the "Default apps" setting and manually reassigning VSCode as default for all supported extensions. Why this helps, I am not quite sure but it works.

**Comment by RyannnnnnR** [comment]
Same issue as @brezanac except resetting the file extensions didn't fix my issue. I noticed that when I went to associate the file types by right clicking on the file, the visual studio code had a dark background. However, when I went to associate the php files, there was no background on the visual studio code icon.  This feature works for all other file types except php. It just shows the Visual Studio code Icon. Is it possible that these files are associated to an older version of VS code?

![bug](https://user-images.githubusercontent.com/26314009/46808252-a245e900-cdb7-11e8-8ada-602ddd56b8d2.gif)
